libpod/runtime.go

```diff
@@ -666,6 +666,11 @@ func (r *Runtime) TmpDir() (string, error) {
 	return r.config.Engine.TmpDir, nil
 }
 
+func (r *Runtime) NewOCIRuntimeFromPath(name, path string) (OCIRuntime, error) {
```

Comment:
This should *not* be exposed outside of Libpod. We can add the WASM+crun config to `containers.conf` or have a bool for the runtime to create an extra copy that supports WASM. We should not do any updates to the runtime from outside Libpod.

Reply:
In future we might have more runtimes and we create the runtime before we get `image` data, pre-creating runtimes prehand will use unnecessary memory if they are not being used by the image. My idea was to only create runtime if its needed otherwise not but I'm open if there is any better approach.

Reply:
a new API could be just `CreateAndSwapRuntime("crun-wasm")` and specgen just calls this and libpod internally does everything. WDYT ?

Reply:
I still see no reason why we can't create the extra runtime struct, and then Specgen just requests the `crun-wasm` runtime. Less code, simpler.

Reply:
To be completely clear, we can already have up to 5 runtimes (there are 5 defined in default containers.conf). I see no reason why a 6th is any issue.

Reply:
This is fixed.